lib/access-context.js

```diff
@@ -126,8 +128,20 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    // the principalType must either be 'USER' or permit to resolve a valid user model
     if (p.type === Principal.USER) {
       return p.id;
+    } else {
```

Comment:
well it would do no arm given the following tests on base model name, but i added a new test: 
```
if (p.type === Principal.ROLE || p.type === Principal.ROLE) {
      return null;
    }
```